flash/data/data_pipeline.py

```diff
@@ -0,0 +1,224 @@
+            moved_to_device = func(*args, **kwargs)
+            return collater(moved_to_device)
+
+        return new_func
+
+    def _attach_to_model(self, model: LightningModule, loader_stage: str = 'all') -> LightningModule:
+        if loader_stage == 'all':
+            loader_stage = ['train', 'test', 'val', 'predict']
+
+        elif isinstance(loader_stage, str):
+            loader_stage = [loader_stage]
+
+        for stage in loader_stage:
+            loader_name = f'{stage}_loader'
+
+            if hasattr(model, loader_name):
+                dataloader = getattr(model, loader_name)
+
+                if isinstance(dataloader, _PatchDataLoader):
+                    wrap_patch_loader = True
+                    dataloader = dataloader()
+
+                else:
+                    wrap_patch_loader = False
+
+                if isinstance(dataloader, Sequence):
+                    was_seq = True
+                else:
+                    dataloader = [dataloader]
+                    was_seq = False
+
+                for idx, loader in enumerate(dataloader):
+                    if isinstance(loader, DataLoader):
+                        dl_args = {k: v for k, v in vars(loader).items() if not k.startswith("_")}
+
+                        dl_args['collate_fn'], device_collater = self.split_around_collate(
+                            collate_fn=dl_args['collate_fn']
+                        )
+
+                        loader = type(loader)(**dl_args)
+
+                    dataloader[idx] = loader
+
+                if not was_seq:
+                    dataloader = dataloader[0]
+
+                if wrap_patch_loader:
+                    dataloader = _PatchDataLoader(dataloader)
+
+                setattr(model, loader_name, dataloader)
+
+        model.transfer_batch_to_device = (
+            self._model_transfer_to_device_wrapper(model.transfer_batch_to_device, device_collater)
+        )
+        return model
+
+    def _generate_auto_dset(self, data: Union[Iterable, Any]) -> AutoDataset:
+        if isinstance(data, Iterable) and self.is_overriden('load_sample'):
+            load_per_sample = True
+            load_fn = self.load_sample
+        else:
+            load_per_sample = False
+            load_fn = self.load_data
+
+        return AutoDataset(data=data, load_fn=load_fn, load_per_sample=load_per_sample)
+
+    def _generate_loader(
+        self, data: Union[Iterable, Any], auto_collate: Optional[bool] = None, **loader_kwargs
+    ) -> DataLoader:
+        if 'collate_fn' in loader_kwargs:
+            if auto_collate is not None:
+                raise MisconfigurationException('auto_collate and collate_fn are mutually exclusive')
+
+        else:
+            if auto_collate is None:
+                auto_collate = True
+
+            if auto_collate:
+                loader_kwargs['collate_fn'] = default_collate
+            else:
+                loader_kwargs['collate_fn'] = default_convert
```

Comment:
Isn't `default_convert` used only for numpy array ?

Reply:
No, default convert also converts numbers etc. basically does the same as default_collate without tensor stacking